### PR TITLE
MapControls: support ctrl-drag

### PR DESCRIPTION
**Pull request by WestLangley**
With this PR, left mouse pans, and ctrl/meta + left mouse orbits. This is similar to the Google Maps API.

### CODE REVIEW

**Comment by WestLangley** [comment]
Live: https://rawgit.com/westlangley/three.js/dev-map_controls_ctrl_drag/examples/misc_controls_map.html

**Review by moroine — APPROVED** [review]
Works perfectly on my ubuntu with mouse and also in Mac with trackpad !

File: examples/js/controls/MapControls.js [hunk 1]
```diff
@@ -776,17 +775,29 @@ THREE.MapControls = function ( object, domElement ) {
 
 		switch ( event.button ) {
 
-			case scope.mouseButtons.ORBIT:
+			case scope.mouseButtons.LEFT:
 
-				if ( scope.enableRotate === false ) return;
+				if ( event.ctrlKey || event.metaKey ) {
```
Comment (haroldiedema) [inline examples/js/controls/MapControls.js]: You should handle the control key separately. Right now when you press the ctrl key while already dragging the mouse, it doesn't change its state to rotate. Similarly, it does not stop rotating when you release the ctrl key while you keep dragging the mouse around.

Comment (WestLangley) [reply examples/js/controls/MapControls.js]: >when you press the ctrl key while already dragging the mouse, it doesn't change its state to rotate.

Why would we _prefer_ that behavior?

As currently implemented, the ctrl key must be pressed first.

Comment (mrdoob) [reply examples/js/controls/MapControls.js]: >> when you press the ctrl key while already dragging the mouse, it doesn't change its state to rotate.

> Why would we prefer that behavior?

That's how Google Maps behaves... 🤔

Comment (mrdoob) [reply examples/js/controls/MapControls.js]: Doesn't need to be addressed in this PR though...

Comment (WestLangley) [reply examples/js/controls/MapControls.js]: This can be merged, then. It maintains a similar API between `MapControls` and `OrbitControls`.

**Comment by mrdoob** [comment]
Thanks!

File: examples/js/controls/MapControls.js (hunk 1 above)
Comment (moroine) [reply examples/js/controls/MapControls.js]: Yeah, I think we should keep Orbit & Map core function as close as possible

**Pull request merged**